### PR TITLE
feat(engine-update): Node engine update, ES6 linting enabled, dependencies update

### PR DESCRIPTION
**Pull request by imranzaidi**
## Commit Message Guidelines
```
Header
Blank Line
Body
Blank Line
Footer

The header should look like:
<type>(<scope>): <subject>

The body should have any necessary detailed info about the commit:
An example, references as to where this idea came from, etc.

The footer should have all the issues tagged:
Fixes #123, Fixes #456

So a commit should look like:
feat(users): Add new Yahoo authentication

Yahoo authentication idea proposed by @codydaig
Example implementation in file.js

Fixes #82
```

* Types: 
  * feat - Features, Enhancements, and overall Improvements
  * fix - Fixes, Bugs, HotFixs, etc...
  * doc - Changes to the Documentation and doesn't actually touch any code.
* Scope:
  * The scope should be where the change took place.
  * Examples: users, core, config, articles
* Subject:
  * The subject line should be clear and concise as to what is being accomplished in the commit.
* General Rules:
  * No Line in the Commit message can be longer than 80 characters.
* Reference: [Angular Conventions](https://github.com/ajoslin/conventional-changelog/blob/master/conventions/angular.md)

- updated engine to use node v4.4.7 and npm v3.10.5
- updated .eslinterc to allow ES6
- updated most dependencies to bring them up to date, with out breaking
tests
- added npm script convenience command to clear installed dependencies
(“npm run clean”) and reinstall them (“npm run reinstall”)

### CODE REVIEW

**Comment by imranzaidi** [comment]
After changing gulp-wiredep to wiretap in the second commit, Travis builds for Node 4 & 5 are both passing.

**Comment by imranzaidi** [comment]
Only issue on Node 6 build servers is trivial,

`
  106 passing (6s)
  1 failing
  1) User CRUD tests should be able to login successfully and logout successfully:
      Uncaught AssertionError: expected 'Found. Redirecting to /' to be 'Moved Temporarily. Redirecting to /'
      + expected - actual
      -Found. Redirecting to /
      +Moved Temporarily. Redirecting to /
`

File: package.json [hunk 1]
```diff
@@ -11,11 +11,13 @@
     "url": "https://github.com/meanjs/mean.git"
   },
   "engines": {
-    "node": ">=0.12.0",
-    "npm": ">=2.0.0"
+    "node": "4.4.7",
```
Comment (ilanbiala) [inline package.json]: should be >=

File: package.json [hunk 2]
```diff
@@ -11,11 +11,13 @@
...
+    "npm": "3.10.5"
```
Comment (ilanbiala) [inline package.json]: should be >=

File: package.json [hunk 3]
```diff
@@ -32,19 +34,19 @@
...
     "compression": "~1.6.1",
     "connect-flash": "~0.1.1",
-    "connect-mongo": "~1.1.0",
+    "connect-mongo": "~1.3.1",
     "consolidate": "~0.14.0",
     "cookie-parser": "~1.4.1",
     "crypto": "0.0.3",
     "express": "~4.14.0",
-    "express-session": "~1.13.0",
+    "express-session": "^1.14.0",
```
Comment (ilanbiala) [inline package.json]: ~

File: package.json [hunk 4]
```diff
@@ -71,10 +73,11 @@
     "passport-paypal-openidconnect": "~0.1.1",
     "passport-twitter": "~1.0.4",
     "phantomjs-prebuilt": "~2.1.4",
+    "run-sequence": "^1.1.5",
```
Comment (ilanbiala) [inline package.json]: already in devDependencies

File: package.json (hunk 1 above)
Comment (imranzaidi) [inline package.json]: Fixed in commit 43a9c8aa1c0f11a41559e13f89d9f9dcafb93603

File: package.json (hunk 2 above)
Comment (imranzaidi) [inline package.json]: Fixed in commit 43a9c8aa1c0f11a41559e13f89d9f9dcafb93603

File: package.json (hunk 3 above)
Comment (imranzaidi) [inline package.json]: Fixed in commit 43a9c8aa1c0f11a41559e13f89d9f9dcafb93603

File: package.json (hunk 4 above)
Comment (imranzaidi) [inline package.json]: Fixed in commit 43a9c8aa1c0f11a41559e13f89d9f9dcafb93603

**Comment by ilanbiala** [comment]
@imranzaidi can you take a look at the list @mleanos put here https://github.com/meanjs/mean/issues/1411#issuecomment-235781145 and add that to this PR as well? Once you've done that, then just squash to 1 commit and we should be ready to merge.

**Comment by mleanos** [comment]
@ilanbiala My comment, that you referenced, was pertaining to the issue of not being able to deploy properly to Heroku. I'm not sure if it's relevant to this PR.

**Comment by imranzaidi** [comment]
@ilanbiala I've moved the dependencies mentioned in @mleanos post and squashed the commits.

**Comment by mleanos** [comment]
@imranzaidi Why are you moving those dependencies from `devDependencies`? 

I think my comment was misunderstood. I was providing a solution for thise that needed a workaround for the limitation of the Heroku build process.

**Comment by imranzaidi** [comment]
@mleanos Got it, I'll revert those changes.

**Comment by ilanbiala** [comment]
@mleanos LGTY then?

**Comment by lirantal** [comment]
LGTM, @ilanbiala please squash and merge?

**Comment by mleanos** [comment]
LGTM.

@ilanbiala Sorry I didn't respond to you. I lost track of this one.

File: package.json [hunk 5]
```diff
@@ -11,11 +11,13 @@
...
   "engines": {
-    "node": ">=0.12.0",
-    "npm": ">=2.0.0"
+    "node": ">=4.4.7",
+    "npm": ">=3.10.5"
   },
   "scripts": {
     "update": "npm update && npm prune && bower install --allow-root && bower prune --allow-root",
+    "clean": "rm -rf node_modules/ public/lib/",
+    "reinstall": "npm run clean; npm install",
```
Comment (codydaig) [inline package.json]: shouldn't this be `npm run clean && npm install`

`&&` instead of `;`

**Comment by lirantal** [comment]
@ilanbiala ?

File: package.json (hunk 5 above)
Comment (lirantal) [inline package.json]: @codydaig I think you're right.

**Comment by mleanos** [comment]
@meanjs/contributors This would be nice to get for 0.5.0. Can we make an effort to get this merged?

@lirantal Could you take over this PR? Since you've been more involved in the updating of the packages, I figure you'd have a much better handle on this one.

**Comment by lirantal** [comment]
Sure I'll take a look @mleanos  but we need rebasing there. @imranzaidi are you up for the task?

**Comment by sujeethk** [comment]
@lirantal Last commit on this PR is **3 months old**. Current **master already has higher versions** of the packages this PR is purposed to update. And it would be better if the node engine is upgraded to **v4.6.0** instead of v4.4.7 since 4.6.0 is LTS and contains recent high severity security patches. Infact node v6.x is scheduled to be LTS in less than a week.

The only upgrade from this PR now would be setting es6 to true and the 2 scripts - clean and reinstall(with the modification of && instead of ; )

per david-dm only out of date dependencies are a couple of dev dependencies-
```
eslint
eslint-config-airbnb
```

**Comment by lirantal** [comment]
Thanks @sujeethk, I'll open a new PR with the relevant changes remaining.
Appreciate the comment here!

**Comment by imranzaidi** [comment]
Sorry guys, had my head deep in work and just saw this.

**Pull request closed without merging**